File: src/main/scala/org/xbery/artbeams/common/access/domain/EntityAccessCountFilter.scala
package org.xbery.artbeams.common.access.domain

/**
  * @author Radek Beran
  */
case class EntityAccessCountFilter(entityKey: Option[EntityKey], entityTypeIn: Option[Seq[String]], entityIdIn: Option[Seq[String]])

object EntityAccessCountFilter {
  lazy val Empty = EntityAccessCountFilter(None, None, None)
}

File: src/main/kotlin/org/xbery/artbeams/common/access/domain/EntityAccessCountFilter.kt
package org.xbery.artbeams.common.access.domain

/**
 * @author Radek Beran
 */
data class EntityAccessCountFilter(val entityKey: EntityKey?, val entityTypeIn: List<String>?, val entityIdIn: List<String>?) {
    companion object {
        val Empty: EntityAccessCountFilter = EntityAccessCountFilter(null, null, null)
    }
}
